src/MessageHandler/Consumer.php

```diff
@@ -169,9 +171,16 @@ public function unackAll(): int
         $nbMessageUnack = 0;
 
         do {
-            $message = $this->redisClient->rpoplpush($this->getWorkingList(), $queueList);
-            $nbMessageUnack++;
-        } while (!is_null($message));
+            $message = $this->redisClient->rpop($this->getWorkingList());
+
+            if (!empty($message)) {
+                $messageEnvelope = MessageEnvelope::unserializeMessage($message);
+
+                $this->unack($messageEnvelope);
```

Comment:
We should lpush / incrementRetry manually here because removeMessageInWorkingList uses lrem which iterates on all elements of the list (for nothing here because you called rpop just before)